ui/src/app/store/domain/slice.ts

```diff
@@ -162,6 +164,66 @@ const domainSlice = createSlice({
     ) => {
       state.active = action.payload?.id || null;
     },
+    createRecord: {
```

Comment:
@Caleb-Ellis just pointed this out (I totally missed it) but this should be split into two different calls, one for "createAddressRecord" and one for "createDNSData". It would also mean updating the `domainActions.createRecord` to first check for the record type before it calling the right create method.